beacon_node/eth2-libp2p/src/rpc/methods.rs

```diff
@@ -9,7 +9,16 @@ use types::{Epoch, EthSpec, Hash256, SignedBeaconBlock, Slot};
 
 /* Requests */
 
-pub type RequestId = usize;
+/// Identifier of a request.
+///
+// NOTE: The handler stores the `RequestId` to inform back of responses and errors, but it's execution
+// is independent of the contents on this type.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum RequestId {
```

Comment:
Yeah, this is a good solution. I have been thinking to make this whole headache go away to just use random numbers. 
But I think what you have is good. It's clear. 
After all the changes, its just Sync that actually cares about RPC responses and this clarifies that behaviour.